### PR TITLE
Added alert and recoding rule for ceph node down.

### PR DESCRIPTION
**Pull request by anmolsachan**
ceph-mixins-issue: #36 
Signed-off-by: Anmol Sachan <anmol13694@gmail.com>

### CODE REVIEW

**Comment by shtripat** [comment]
@anmolsachan Fix the build issues

File: alerts/state.libsonnet [hunk 1]
```diff
@@ -36,6 +36,22 @@
               severity_level: 'warning',
             },
           },
+          {
+            alert: 'CephNodeDown',
```
Comment (shtripat) [inline alerts/state.libsonnet]: Lets keep this in `node.libsonnet` as `state.libsonnet` meant for cluster states.

File: extras/manifests/prometheus-rules.yaml [hunk 2]
```diff
@@ -93,6 +98,18 @@ spec:
       for: 10m
       labels:
         severity: warning
+    - alert: CephNodeDown
+      annotations:
+        description: Ceph Storage node {{ $labels.node }} went down. Please check
```
Comment (shtripat) [inline extras/manifests/prometheus-rules.yaml]: Remove the word `Ceph` from description.

File: extras/manifests/prometheus-rules.yaml [hunk 3]
```diff
@@ -93,6 +98,18 @@ spec:
...
+          the node immeditely.
```
Comment (shtripat) [inline extras/manifests/prometheus-rules.yaml]: s/immeditely/immediately/g

File: alerts/state.libsonnet (hunk 1 above)
Comment (umangachapagain) [reply alerts/state.libsonnet]: +1

File: rules/rules.libsonnet [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  prometheusRules+:: {
+    groups+: [
+      {
+        name: 'ceph.rules',
+        rules: [
+          {
+            record: 'ceph_node_down',
```
Comment (umangachapagain) [inline rules/rules.libsonnet]: we need to name this according to the convention.

Comment (umangachapagain) [reply rules/rules.libsonnet]: rules.libsonnet should ideally have only the imports. This rule should be in some other file, say 'node.libsonnet'

File: rules/rules.libsonnet [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+            expr: |||
+              kube_node_status_condition{condition="Ready",job="kube-state-metrics",status="true"} * on (node) group_right() max(label_replace(ceph_disk_occupation,"node","$1","exported_instance","(.*)")) by (node) == 0
```
Comment (umangachapagain) [inline rules/rules.libsonnet]: ceph_disk_occupation{job="rook-ceph-mgr"}

**Review by umangachapagain — CHANGES_REQUESTED** [review]

**Comment by umangachapagain** [comment]
Fix Lint issues.

File: extras/manifests/prometheus-rules.yaml (hunk 2 above)
Comment (anmolsachan) [reply extras/manifests/prometheus-rules.yaml]: ack

File: extras/manifests/prometheus-rules.yaml (hunk 3 above)
Comment (anmolsachan) [reply extras/manifests/prometheus-rules.yaml]: ack

File: rules/rules.libsonnet (hunk 4 above)
Comment (anmolsachan) [reply rules/rules.libsonnet]: ack, will change

File: rules/rules.libsonnet (hunk 5 above)
Comment (anmolsachan) [reply rules/rules.libsonnet]: ack

**Review by shtripat — APPROVED** [review]
LGTM

**Review by umangachapagain — APPROVED** [review]

**Pull request merged**